### PR TITLE
Metadata filtering tutorial

### PR DESCRIPTION
**Pull request by TuanaCelik**
This adds the beginner tutorial for metadata filtering, to try, you can also use this colab: https://colab.research.google.com/drive/1mx3Wk3nUA0zGLmeqZSSgZzQBbMFUitOJ?usp=sharing

### CODE REVIEW

File: index.toml [hunk 1]
```diff
@@ -296,4 +296,15 @@ notebook = "29_Serializing_Pipelines.ipynb"
 aliases = []
 completion_time = "10 min"
 created_at = 2024-01-29
+haystack_2 = true
+
+[[tutorial]]
+title = "Filtering Documents with Metadata"
+description = "Learn how to filter down to specific documents at retieval time using metadata"
```
Comment (bilgeyucel) [inline index.toml:303]:
```suggestion
description = "Learn how to filter down to specific documents at retrieval time using metadata"
```

**Review by bilgeyucel — APPROVED** [review]

File: tutorials/31_Metadata_Filtering.ipynb [hunk 2]
```diff
@@ -0,0 +1,966 @@
+{
```
Comment (bilgeyucel) [inline tutorials/31_Metadata_Filtering.ipynb:1]: <p>Do all document stores support metadata filtering? We can add this as a warning here</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/deepset-ai/haystack-tutorials/pull/275/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='7'/>

Comment (TuanaCelik) [reply tutorials/31_Metadata_Filtering.ipynb:1]: Good question, I'll check

Comment (TuanaCelik) [reply tutorials/31_Metadata_Filtering.ipynb:1]: They all implement it, pgvector in development

**Pull request merged**